packages/schematics/src/command-line/shared.ts

```diff
@@ -345,6 +345,12 @@ export function getAllLibNames() {
   return getProjectNames(p => p.type === ProjectType.lib);
 }
 
+export function getAllProjectNames() {
+  return getProjectNames(
+    p => p.type === ProjectType.app || p.type === ProjectType.lib
```

Comment:
Hm is there a particular reason why **E2E** projects are not in this list?

I imagine this command is used mostly for display purposes, it seems similar to `yarn affected:dep-graph` but less graphical. Dep graph also shows `E2E`.

We also refer to E2E code as projects... If this only returns apps and libs, it might be confusing to call it `affected:projects`  and not return E2E projects.

Otherwise, you can use `getProjectNames()`

Reply:
For the issue I'm addressing, the user was only concerned about apps and libs. In my work's project, I wrote a script that runs commands on affected apps/libs that have changed since diverging from a given parent branch. I do it like this:

```
const affectedAppsOutput: string = execSync(`yarn affected:apps --base=${parentBranch}`).toString();
const affectedLibsOutput: string = execSync(`yarn affected:libs --base=${parentBranch}`).toString();
```

I don't care about E2E projects that have been changed since that wouldn't affect anything I care about.

The goal is to condense this down into a single command, I think.

If we didn't want to call this `affected:projects`, what would you suggest?

Reply:
Hm, I guess I'm unsure what the end goal is.

We have `affected --target` which is generic enough to run any builder. We also have a new `run-commands` builder which will run any `sh` script... With these I think it's pretty much possible to do many things.

Sorry, I thought this was for all projects which makes sense to me for display purposes. Can we go back to the issue and discuss a bit more?

Reply:
For me the end goal is to get all affected projects from angular.json in one command. 
`affected --target` get the string but we can not pass apps and libs.

In my case affected project give me easy access to root path from angular.json and then collect all test cases.